packages/devtools_app/lib/src/screens/vm_developer/object_inspector_view_controller.dart

```diff
@@ -31,6 +32,8 @@ class ObjectInspectorViewController extends DisposableController
   final programExplorerController =
       ProgramExplorerController(showCodeNodes: true);
 
+  final debuggerController = DebuggerController();
```

Comment:
shouldn't we reuse the same debuggerController as the debugger page using the provider? Otherwise I imagine things like setting and getting breakpoints could be strangely inconsistent.

Reply:
Nope, because then updating the current script in the debugger on either screen will result in a change on the other. Setting and getting breakpoints seems to work fine since the controller is wired in such a way that it will react to new breakpoints being added / removed by other clients. Seemed to work fine with manual testing.

Reply:
Ideally, we'd split `DebuggerController` into two components: `DebuggerController` and `CodeViewController`, where `CodeViewController` is responsible for responsible for script state and jumping to locations in the script. That way, `DebuggerController` would only be responsible for debugger state and could be reused across the application.

Reply:
Yeah please split into a separate CodeViewController.  That should be pretty easy.
FWIW, I'd be ok with landing even if you cause the script to change on the other page. That is actually a bit of a feature as well as a bug. E.g. I was viewing this file in the program explorer and now I see it in the debugger page... convenient.

Reply:
Eh, I'm not convinced that changing the script in a separate tab as a result of operations performed in another is good UX unless we have an explicit way to jump to the other tab displaying the updated script. These kind of surprises will cause more confusion than anything.

Reply:
ok fine to leave as a todo and add a CodeViewController later

Reply:
... too late :-)